source/common/buffer/zero_copy_input_stream_impl.cc

```diff
@@ -0,0 +1,54 @@
+#include "common/buffer/zero_copy_input_stream_impl.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Buffer {
+
+ZeroCopyInputStreamImpl::ZeroCopyInputStreamImpl(Buffer::Instance& buffer) {
+  move(buffer);
+  finish();
+}
+
+void ZeroCopyInputStreamImpl::move(Buffer::Instance& instance) {
+  ASSERT(!finished_);
+
+  buffer_.move(instance);
+}
+
+bool ZeroCopyInputStreamImpl::Next(const void** data, int* size) {
+  if (position_ != 0) {
+    buffer_.drain(position_);
+    position_ = 0;
+  }
+
+  Buffer::RawSlice slice;
+  const uint64_t num_slices = buffer_.getRawSlices(&slice, 1);
```

Comment:
See comment here: https://github.com/lyft/envoy/blob/master/include/envoy/buffer/buffer.h#L67 I think this can return a slice that has no data in it. Just make sure the code below correctly handled that case. I think you probably need to also check that slice.len_ > 0 but not sure.